### PR TITLE
Add `VirtualForm#errors`

### LINKED ISSUE #29: Add VirtualForm#errors method similar to the one available for regular forms

### PR DESCRIPTION
**Pull request by HarrisonB**
Closes #29.

### CODE REVIEW

File: src/avram/form_errors.cr [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  def errors
+    fields.reduce({} of Symbol => Array(String)) do |errors_hash, field|
+      if field.errors.empty?
+        errors_hash
+      else
+        errors_hash[field.name] = field.errors
+        errors_hash
+      end
+    end
+  end
```
Comment (paulcsmith) [inline src/avram/form_errors.cr]: Can you add an `abstract def fields` to this just so it is extra clear what this module requires?

File: src/avram/form_errors.cr [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module Avram::FormErrors
+  def errors
```
Comment (paulcsmith) [inline src/avram/form_errors.cr]: Could you add a return type here? If you're not sure what it is you can make it String and then the compiler will tell you what it actually should be

**Comment by paulcsmith** [comment]
Also looks like it needs to be formatted: `crystal tool format src spec config`

**Comment by HarrisonB** [comment]
Should be all good 😄

**Pull request merged**